Ch. 17: minor typos and link reference (#4286)
rust-lang/book: src/ch17-01-futures-and-syntax.md

}
```

> The `race` function returns `Left` with that future’s output if the first argument
> wins, and `Right` with the second future's output if the second argument wins. This
> matches the order the arguments appear in when calling the function: the first
> argument is to the left of the second argument.

Reviewer: Good catch that there’s some ambiguity here. I think we actually need a slightly more substantial rephrasing to make it less clunky than what I originally wrote.

```suggestion
The `race` function returns `Left` with the output from the first future
argument it finishes first, or `Right` with the output of the second future
argument if that one finishes first. This matches the order the arguments appear
in when calling the function: the first argument is to the left of the second
argument.
```
